server/src/main/java/org/elasticsearch/cluster/ClusterModule.java

```diff
@@ -172,6 +173,9 @@ public static List<Entry> getNamedWriteables() {
 
         // Task Status (not Diffable)
         entries.add(new Entry(Task.Status.class, PersistentTasksNodeService.Status.NAME, PersistentTasksNodeService.Status::new));
+
+        // Health node selector task is always present in the cluster state
+        entries.addAll(HealthNodeSelectorTaskExecutor.getNamedWriteables());
```

Comment:
I think we should only register this if the feature flag is enabled

Reply:
Agreed.